substrate/in-instructions/primitives/src/incoming.rs

```diff
@@ -0,0 +1,52 @@
+use scale::{Encode, Decode, MaxEncodedLen};
+use scale_info::TypeInfo;
+
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
+
+use sp_core::{ConstU32, bounded::BoundedVec};
+
+use serai_primitives::NativeAddress;
+
+use crate::{MAX_DATA_LEN, ExternalAddress};
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Application {
```

Comment:
This seems unnecessary at this stage, not sure that we would define applications under an enum anyways?

Reply:
It is unnecessary, yet if we ever write anything distinct, this allows us to do so in a non-breaking way for the cost of a single byte.

Without this, we'd need to do a breaking change in format if we ever offer any additional services. While I'm not here to harp for adding back a SC VM right now, or say we're shipping lending, I just figured this was the path offering the least resistance.

I believe using an enum absolutely makes sense though.